Assets/Scripts/Stats.cs

```diff
@@ -17,24 +17,49 @@ public Stats(float maxHealth = 100f)
     void Start()
     {
         // TODO: refactor for more dynamic assignment
-        if (gameObject.tag == "Player") owner = EntityType.PLAYER;
+        if (gameObject.tag == "Player")
+        {
+            owner = EntityType.PLAYER;
+            UIManager.MaxHealth = maxHealth;
+        }
         else owner = EntityType.ENEMY;
-        maxHealth = 100f;
-        currentHealth = maxHealth;
+
+        setHealth(maxHealth);
+    }
+
+    void setHealth(float health)
+    {
+        currentHealth = Mathf.Clamp(health, 0, maxHealth);
+        if (owner == EntityType.PLAYER)
+        {
+            UIManager.Health = currentHealth;
+            SpeedManager.updateSpeeds(currentHealth / maxHealth);
+        }
     }
 
-    // Update is called once per frame
-    void Update()
+    void setMaxHealth(float health)
     {
+        maxHealth = health;
         currentHealth = Mathf.Clamp(currentHealth, 0, maxHealth);
-        if(owner == EntityType.PLAYER) {
+        if (owner == EntityType.PLAYER)
+        {
+            UIManager.MaxHealth = maxHealth;
             SpeedManager.updateSpeeds(currentHealth / maxHealth);
         }
     }
 
+    public void heal(float healthGained)
+    {
+        setHealth(currentHealth + healthGained);
+        if (currentHealth <= 0)
+        {
+            Destroy(gameObject);
+        }
+    }
+
     public void takeDamage(float damage)
     {
-        currentHealth -= damage;
+        setHealth(currentHealth - damage);
```

Comment:
we can revisit the structure of heal and takeDamage, they're pretty much the same method